rig/place_and_route/route/ner.py

```diff
@@ -0,0 +1,436 @@
+"""Neighbour Exploring Routing (NER) algorithm from J. Navaridas et al.
+
+Algorithm refrence: J. Navaridas et al. SpiNNaker: Enhanced multicast routing,
+Parallel Computing (2014).
+
+`http://dx.doi.org/10.1016/j.parco.2015.01.002`
+"""
+
+import heapq
+
+from collections import deque
+
+from .util import longest_dimension_first, to_xyz, \
+    shortest_mesh_path_length, shortest_mesh_path, \
+    shortest_torus_path_length, shortest_torus_path, \
+    has_wrap_around_links, links_between
+
+from ..exceptions import MachineHasDisconnectedSubregion
+
+from ..constraints import RouteEndpointConstraint
+
+from ...machine import Links, Cores
+
+from ...routing_table import Routes
+
+from ..routing_tree import RoutingTree
+
+
+def concentric_hexagons(radius, start=(0, 0)):
```

Comment:
There are quite a large number of geometry-related utility functions floating around. Other pertinent examples are the shortest-path and distance calculation functions. That and "hey tell me how a board is laid out" for example.

I suppose a root-level geometry module would be appropriate. I'll flag this for refactoring soon but not this PR.